src/main/java/common/DateParser.java

```diff
@@ -0,0 +1,88 @@
+package common;
+
+import storage.Storage;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.util.Locale;
+
+/**
+ * The DateParser class.
+ * Handles parsing date strings to {@link LocalDateTime}
+ * and displaying {@link LocalDateTime} as "MMM dd yyyy h:mma".
+ * Example: Aug 10 2023 6:00PM.
+ */
+public class DateParser {
+    /**
+     * The {@link DateTimeFormatter} used to parse
+     * date strings into {@link LocalDateTime}.
+     */
+    private static final DateTimeFormatter parseFormat = 
+        new DateTimeFormatterBuilder()
+            .append(DateTimeFormatter.ofPattern(
+                "[d/M/yyyy]"
+                    + "[yyyy/M/d]"
+                    + "[d-M-yyyy]"
+                    + "[yyyy-M-d]"
+            ))
+            .append(DateTimeFormatter.ofPattern(
+                "[d MMM yyyy]" + "[MMM d yyyy]",
+                Locale.ENGLISH
```

Comment:
Good use of Locale.ENGLISH to ensure consistency in date formats regardless of the system's locale.